src/overlays/actors/ovl_Bg_Tobira01/z_bg_tobira01.c

```diff
@@ -28,14 +28,79 @@ const ActorInit Bg_Tobira01_InitVars = {
     (ActorFunc)BgTobira01_Update,
     (ActorFunc)BgTobira01_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/func_80B12430.s")
+void BgTobira01_Action(BgTobira01* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s16 csId = this->dyna.actor.csId;
+    s16 tempPosY;
+
+    if (this->cutsceneRequested) {
+        if (CutsceneManager_GetCurrentCsId() == CS_ID_GLOBAL_TALK) {
+            CutsceneManager_Stop(CS_ID_GLOBAL_TALK);
+        } else if (CutsceneManager_IsNext(csId)) {
+            CutsceneManager_StartWithPlayerCs(csId, &this->dyna.actor);
+            SET_WEEKEVENTREG(WEEKEVENTREG_88_40);
+            this->cutsceneRequested = false;
+        } else {
+            CutsceneManager_Queue(csId);
+        }
+    } else if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40) && (this->posYTick == 0) &&
+               (play->actorCtx.playerImpact.timer != 0) &&
+               (play->actorCtx.playerImpact.type == PLAYER_IMPACT_GORON_GROUND_POUND) &&
+               (SurfaceType_GetSceneExitIndex(&play->colCtx, player->actor.floorPoly, player->actor.floorBgId) == 6)) {
+        this->cutsceneRequested = true;
+        this->unk16C = 0;
+    }
+
+    tempPosY = this->posYTick;
```

Comment:
```suggestion
    prevPosYTick = this->posYTick;
```
Usually when a variable is stored when it's updated, the convention is `prev` followed by the original variable name